### PR TITLE
Evm contract call from vm test

### PR DESCRIPTION
**Pull request by cwarnerdev**
- adds a test to call magic contract method from within vm.
- fixes a nil pointer edge case
- disabled some evmtest code that is not working. will investigate for further improvement/fixes

### CODE REVIEW

File: packages/vm/vmimpl/call.go [hunk 1]
```diff
@@ -43,10 +43,12 @@ const traceStack = false
...
-		contract:           contract,
-		params:             args,
-		allowanceAvailable: allowance.Clone(), // we have to clone it because it will be mutated by TransferAllowedFunds
+		caller:   caller,
+		contract: contract,
+		params:   args,
+	}
+	if allowance != nil {
+		ctx.allowanceAvailable = allowance.Clone() // we have to clone it because it will be mutated by TransferAllowedFunds
 	}
```
Comment (dessaya) [inline packages/vm/vmimpl/call.go]: We changed `Assets` so that it should never be nil. So instead of checking here it is better to find the root cause of why it is nil and fix it there.

**Comment by cwarnerdev** [comment]
No more nil check for the allowance :tada:

**Review by dessaya — APPROVED** [review]
